PocketKit/Sources/PocketKit/Premium/PremiumUpgradeView.swift

```diff
@@ -0,0 +1,228 @@
+                )
+            }
+        }
+    }
+}
+
+private struct PremiumYearlyPercent: View {
+    var body: some View {
+        VStack {
+            Text("Save 25%")
+                .style(.percentSaved)
+                .multilineTextAlignment(.center)
+        }
+        .frame(width: 60, height: 60, alignment: .center)
+        .background(Circle().fill(Color(.ui.teal3)))
+    }
+}
+
+private struct PremiumInfoView: View {
+    private let text = """
+Subscriptions will be charged to your credit card through your iTunes account. \
+Your account will be charged $TBD (monthly) or $TBD (yearly) for renewal within 24 hours prior to the end of the current period. \
+Subscriptions will automatically renew unless canceled at least 24 hours before the end of the current period. \
+It will not be possible to immediately cancel a subscription. \
+You can manage subscriptions and turn off auto-renewal by going to your account settings after purchase. \
+Refunds are not available for unused portions of a subscription.
+"""
+
+    var body: some View {
+        Text(text).style(.info)
+    }
+}
+
+private struct PremiumTermsView: View {
+    var body: some View {
+        HStack(spacing: 16) {
+            Button { } label: { Text("Privacy Policy").style(.terms) }
+            Button { } label: { Text("Terms of Service").style(.terms) }
+        }.padding(.bottom)
+    }
+}
+
+private struct PremiumBackgroundView: View {
+    var body: some View {
+        VStack(spacing: 0) {
+            Image(asset: .premiumBorderTop).resizable().frame(maxHeight: borderWidth)
+            sidebars
+            Image(asset: .premiumBorderBottom).resizable().frame(maxHeight: borderWidth)
+        }
+    }
+
+    private var sidebars: some View {
+        HStack(spacing: 0) {
+            Image(asset: .premiumBorderLeft).resizable().frame(maxWidth: borderWidth)
+            Spacer()
+            Image(asset: .premiumBorderRight).resizable().frame(maxWidth: borderWidth)
+        }
+    }
+
+    private var borderWidth: CGFloat {
+        CGFloat(UIDevice.current.userInterfaceIdiom == .pad ? 18.0 : 13.0)
+    }
+}
+
+private extension Style {
+    static let upgradeHeader = Style.header.display.medium.h1.with { style in
+        style.with(alignment: .center)
+    }
+
+    static let featureRow = Style.settings.row.default.with(size: .p3)
+
+    static let monthlyPremiumRow = Style.settings.row.default.with(size: .h4).with(weight: .medium)
+
+    static let yearlyPremiumRow = Style.settings.row.darkBackground.default.with(size: .h4).with(weight: .medium)
+
+    static let pricing = Style.featureRow.with(size: .p4)
+
+    static let yearlyPricing = Style.yearlyPremiumRow.with(size: .p4)
+
+    static let percentSaved = Style.yearlyPremiumRow.with(size: .p3).with(weight: .medium)
+
+    static let info = Style.body.sansSerif.with(size: .p4).with(color: .ui.grey4)
+
+    static let terms = Style.settings.button.default.with(color: .ui.grey4)
```

Comment:
I liked @kybhamilton's structure of defining the Style for a section (see Style+Settings). What do you think about grouping these under `Style+Premium`. We should probably revisit how we do our styling and image assets in the future though.

Reply:
This is strictly related to my lack of ios experience, but what benefit does it have? Also, I'm not entirely sure how to do that.  happy to take some advice on a good starting place 😄